compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Unifier.java

```diff
@@ -557,7 +557,8 @@ public BType visit(BParameterizedType originalType, BType expType) {
                     }
 
                     BType type = paramValueTypes.get(paramVarName);
-                    return type == symbolTable.semanticError ? expType : ((BTypedescType) type).constraint;
+                    return type == symbolTable.semanticError ? expType : 
```

Comment:
Can we check type tag here instead?

Reply:
Yes, we can. But in every other places we have followed this approach.

Reply:
I think we should use tag for new changes since it's the current approach we're using for new additions

Reply:
Fixed.